e2e/node/basic/mainnet.test.ts

```diff
@@ -201,7 +204,15 @@ test('it should allow you to set an incorrect root key', async () => {
     canisterId: Principal.fromText('rrkah-fqaaa-aaaaa-aaaaq-cai'),
   });
 
-  expect(actor.whoami).rejects.toThrowError(`Invalid certificate:`);
+  expect.assertions(3);
+  try {
+    await actor.whoami();
+  } catch (error) {
+    expect(error).toBeInstanceOf(AgentError);
+    const errorCode = (error as AgentError).cause.code as WithRequestDetailsErrorCode;
+    expect(errorCode).toBeInstanceOf(WithRequestDetailsErrorCode);
+    expect(errorCode.caughtErrorCode).toBeInstanceOf(MissingSignatureErrorCode);
```

Comment:
I'm a bit conflicted about the extra layer of indirection... But maybe it depends on the usage. I'm not sure if we will expect devs to interact with the code or not.

Maybe we could remove the indirection by adding optional request details to `AgentError`.